### PR TITLE
feat: add C implementation of `stats/base/dists/rayleigh/logcdf`

### LINKED ISSUE #3801: [RFC]: Add C implementation for `@stdlib/stats/base/dists/rayleigh/logcdf`
### Description

This RFC proposes adding a C implementation, including a Node.js native addon, C benchmarks, and C examples, for the [`@stdlib/stats/base/dists/rayleigh/logcdf`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/@stdlib/stats/base/dists/rayleigh/logcdf) package.

When adding support, the following tasks should be completed:

1. A C source implementation should be added to a `src` folder within the package's directory.
2. A native add-on interface should be added in order to call the C implementation from JavaScript.
3. A JavaScript file should be added to the `lib` folder to allow benchmarking and unit testing the native implementation from JavaScript.
4. A new test file should be created which includes unit tests testing against expected behavior.
5. C benchmarks should be added which measure performance for the C source implementation.
6. A file containing C examples should be added to demonstrate example usage.
7. JavaScript benchmarks should be added which measure performance for the native add-on.
8. Build and configuration files should be added to allow compiling and running the C source implementation using project `make` commands (as documented below).
9. The README should be updated to include documentation for the added C API.

To provide a concrete example of what a PR adding the desired method should contain, see https://github.com/stdlib-js/stdlib/pull/3354, which is a PR adding a C implementation for the CDF of an arcsine distribution. This should provide an idea of what is expected.

#### Prerequisites

1. Review JavaScript implementation in `lib/main.js` and its required modules to identify all `stdlib` functions used.
2. Verify C implementations exist for each required function.

### Related Issues

#3355

### Questions

No.

### Other

Once the implementation is ready including C examples and benchmarks, we should be able to run the following `make` commands.

#### Build native add-on

```bash
NODE_ADDONS_PATTERN="@stdlib/stats/base/dists/rayleigh/logcdf" make install-node-addons
```

#### Run C examples

```bash
make examples-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/rayleigh/logcdf/examples/c/example.c"
```

#### Run C benchmarks

```bash
make benchmark-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/rayleigh/logcdf/benchmark/c/benchmark.c"
```

#### Run JavaScript benchmarks

```bash
make benchmark-javascript-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/rayleigh/logcdf/benchmark/benchmark.native.js"
```

#### Run all tests

```bash
make TESTS_FILTER=".*/@stdlib/stats/base/dists/rayleigh/logcdf/.*" test
```

### Notes

- If you are interested in contributing a PR which addresses this RFC and are a **first-time contributor** or not yet familiar with our project conventions, please minimize the submission of LLM-generated code. Please consult our [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md) and the associated [development guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/development.md). Failure to respect project conventions will result in your PR being rejected without review. Thank you for understanding!

### Checklist

- [x] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [x] Searched for existing issues and pull requests.
- [x] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Vinit-Pandit**
Resolves #3801

## Description

> What is the purpose of this pull request?

This pull request:

-   {{TODO: add description describing what this pull request does}}

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3801 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/logcdf/benchmark/benchmark.js [hunk 1]
```diff
@@ -32,15 +33,22 @@ var logcdf = require( './../lib' );
...
+		x[ i ] = ( randu() * 100.0 ) - 100.0;
+		sigma[ i ] = ( randu() * 20.0 ) + EPS;
+	}
+
 	b.tic();
 	for ( i = 0; i < b.iterations; i++ ) {
-		x = ( randu()*100.0 ) - 100;
-		sigma = ( randu()*20.0 ) + EPS;
-		y = logcdf( x, sigma );
+		y = logcdf( x[i % 100], sigma[ i % 100 ] );
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/logcdf/benchmark/benchmark.js]:
```suggestion
		y = logcdf( x[ i % len ], sigma[ i % len ] );
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/logcdf/src/main.c [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+		stdlib_base_is_nan( x ) ||
+		stdlib_base_is_nan( sigma ) ||
+		sigma < 0.0
+	) {
+		return 0.0/0.0; // NaN
+	}
+	if ( sigma == 0.0 ) {
+		return ( x < 0.0 ) ? STDLIB_CONSTANT_FLOAT64_NINF : 0.0;
+	}
+	if ( x < 0.0) {
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/logcdf/src/main.c]:
```suggestion
	if ( x < 0.0 ) {
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/logcdf/lib/native.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+// MODULES //
+
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Evaluates the logarithm of the cumulative distribution function (CDF) for a Rayleigh distribution with scale parameter `sigma` at a value `x`.
+*
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/logcdf/lib/native.js:30]:
```suggestion

* @private
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/logcdf/lib/native.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+// MODULES //
+
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Evaluates the logarithm of the cumulative distribution function (CDF) for a Rayleigh distribution with scale parameter `sigma` at a value `x`.
+
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/logcdf/lib/native.js]:
```suggestion
*
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/logcdf/benchmark/c/benchmark.c [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+	double elapsed;
+	double x[ 100 ];
+	double sigma[ 100 ];
+	double y;
+	double t;
+	int i;
+
+	for ( i = 0; i < 100; i++ ) {
+		x[ i ] = random_uniform( -100.0, 0.0 );
+		sigma[ i ] = random_uniform( STDLIB_CONSTANT_FLOAT64_EPS , STDLIB_CONSTANT_FLOAT64_EPS + 20 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/logcdf/benchmark/c/benchmark.c]:
```suggestion
		sigma[ i ] = random_uniform( STDLIB_CONSTANT_FLOAT64_EPS , STDLIB_CONSTANT_FLOAT64_EPS + 20.0 );
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/logcdf/src/main.c [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+#include "stdlib/math/base/special/expm1.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/math/base/special/log1p.h"
+#include "stdlib/math/base/special/exp.h"
+#include "stdlib/math/base/special/pow.h"
+#include "stdlib/math/base/special/ln.h"
+#include "stdlib/constants/float64/ln_half.h"
+#include "stdlib/constants/float64/ninf.h"
+
+
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/logcdf/src/main.c]:
```suggestion

```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/logcdf/src/main.c [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+/**
+* Evaluates the logarithm of the cumulative distribution function (CDF) for a Rayleigh distribution with scale parameter `sigma` at a value `x`.
+*
+* @param x        input value
+* @param sigma    scale parameter
+* @return         evaluated LOGCDF
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/logcdf/src/main.c]: To match `native.js` return value description: 

```suggestion
* @return         evaluated logCDF
```

**Review by Planeshifter — APPROVED** [review]
Thank you, @MeastroZI!

**Pull request merged**